lib/lambda_ethereum_consensus/state_transition/operations.ex

```diff
@@ -713,4 +713,87 @@ defmodule LambdaEthereumConsensus.StateTransition.Operations do
     end)
     |> String.length()
   end
+
+  def process_bls_to_execution_change(state, signed_address_change) do
+    address_change = signed_address_change.message
+
+    case validate_address_change(state, address_change) do
+      {:ok, _} ->
+        validator = Enum.at(state.validators, address_change.validator_index)
+
+        case validate_withdrawal_credentials(validator, address_change) do
+          {:ok, _} ->
+            domain =
+              Misc.compute_domain(
+                Constants.domain_bls_to_execution_change(),
+                genesis_validators_root: state.genesis_validators_root
+              )
+
+            signing_root = Misc.compute_signing_root(address_change, domain)
+
+            if Bls.valid?(
+                 address_change.from_bls_pubkey,
+                 signing_root,
+                 signed_address_change.signature
+               ) do
+              new_withdrawal_credentials =
+                Constants.eth1_address_withdrawal_prefix() <>
+                  <<0::size(88)>> <> address_change.to_execution_address
+
+              updated_validators =
+                update_validator_withdrawal_credentials(
+                  state.validators,
+                  address_change.validator_index,
+                  new_withdrawal_credentials
+                )
+
+              {:ok, %BeaconState{state | validators: updated_validators}}
+            else
+              {:error, "bls verification failed"}
+            end
+
+          {:error, msg} ->
+            {:error, msg}
+        end
+
+      {:error, msg} ->
+        {:error, msg}
+    end
+  end
+
+  defp validate_address_change(state, address_change) do
+    if address_change.validator_index < length(state.validators) do
+      {:ok, address_change}
+    else
+      {:error, "Invalid address change"}
+    end
+  end
+
+  defp validate_withdrawal_credentials(validator, address_change) do
+    if binary_part(validator.withdrawal_credentials, 0, 1) == Constants.bls_withdrawal_prefix() and
+         binary_part(validator.withdrawal_credentials, 1, 31) ==
+           binary_part(:crypto.hash(:sha256, address_change.from_bls_pubkey), 1, 31) do
+      {:ok, true}
```

Comment:
We should just return `:ok` here

Reply:
but isn't it a convention to use `{:ok, value}` , `{:error, err_msg}`? If it omit `value` in the `:ok` variant, should I drop the `error_msg` in the `:error` variant to keep consistent?

Reply:
`{:ok, value}` is a convention for returning a value along with the `:ok`. We don't need to return a value if it isn't needed.
Also, we can drop the message for the error variant, but doing so would mean all callers of the function have to handle the specific `:error` case, and that might be cumbersome.

Reply:
ok I changed it.